src/column.rs

```diff
@@ -581,11 +571,21 @@ impl Column {
 				if entry.is_empty() {
 					continue;
 				}
-				let address = entry.address(source.id.index_bits());
-				if skip_preimage_indexes && self.preimage && address.size_tier() != 15 {
+				let (size_tier, offset) = if self.db_version >= 4 {
+					let address = entry.address(source.id.index_bits());
+					(address.size_tier(), address.offset())
+				} else {
+					let addr_bits = source.id.index_bits() + 10;
```

Comment:
```suggestion
					let addr_bits = source.id.index_bits() + CHUNK_ENTRIES_BITS + SIZE_TIERS_BITS - 4;

```

Reply:
This should not be defined in terms of current value of `SIZE_TIERS_BITS`. If `SIZE_TIERS_BITS` changes in the future, this would need to be updated too. I think a hardcoded value is fine here. V3 had exactly  + 10 bits and that's not going to change.